src/compositor/lipstickcompositor.cpp

```diff
@@ -651,11 +668,25 @@ void LipstickCompositor::reactOnDisplayStateChanges(MeeGo::QmDisplayState::Displ
 
     bool changeInDimming = (state == MeeGo::QmDisplayState::Dimmed) != (m_currentDisplayState == MeeGo::QmDisplayState::Dimmed);
 
+    bool changeInAmbient = ((state == MeeGo::QmDisplayState::Off) != (m_currentDisplayState == MeeGo::QmDisplayState::Off)) && ambientEnabled();
+
+    bool enterAmbient = changeInAmbient && (state == MeeGo::QmDisplayState::Off);
+    bool leaveAmbient = changeInAmbient && (state != MeeGo::QmDisplayState::Off);
+
     m_currentDisplayState = state;
 
     if (changeInDimming) {
         emit displayDimmedChanged();
     }
+    if (changeInAmbient) {
+        emit displayAmbientChanged();
```

Comment:
I feel this could be made simpler by adding a bool parameter to displayAmbientChanged that specifies whether we entered Ambient mode or not
`emit displayAmbientChanged(state == MeeGo::QmDisplayState::Off);`

Reply:
Yup you are correct, the reason for having the separate enter and leave event is because there are signals for AboutToBeOn/AboutToBeOff and DisplayOn/DisplayOff, so for convenience and potential future use in watchfaces.